samples/CrawlFeaturizer/Util/CognitiveTextAnalyzer.cs

```diff
@@ -1,62 +1,43 @@
-﻿using Microsoft.Azure.CognitiveServices.Language.TextAnalytics;
-using Microsoft.Azure.CognitiveServices.Language.TextAnalytics.Models;
+﻿using Azure;
+using Azure.AI.TextAnalytics;
 using System.Collections.Generic;
-using System.Linq;
 using System.Threading.Tasks;
 
 namespace CrawlFeaturizer.Util
 {
     public class CognitiveTextAnalyzer
     {
-        private readonly ITextAnalyticsClient textAnalyticsClient = null;
+        private readonly TextAnalyticsClient textAnalyticsClient = null;
 
         /// <summary>
         /// Initializes <see cref="ITextAnalyticsClient"/> object using cognitive service subscription key and endpoint region.
         /// </summary>
         /// <param name="textAnalyticsClient">ITextAnalyticsClient client.</param>
-        internal CognitiveTextAnalyzer(ITextAnalyticsClient textAnalyticsClient)
+        internal CognitiveTextAnalyzer(TextAnalyticsClient textAnalyticsClient)
         {
             this.textAnalyticsClient = textAnalyticsClient;
         }
 
         /// <summary>
-        /// Gets key phrases for the passed input text.
+        /// Gets key phrases for the passed document.
         /// </summary>
-        /// <param name="inputText">Text to analyze</param>
-        public async Task<IList<string>> GetKeyPhrasesAsync(string inputText)
+        /// <param name="document">Text to analyze</param>
+        public async Task<KeyPhraseCollection> GetKeyPhrasesAsync(string document)
         {
-            var response = await GetKeyPhrasesBatchAsync(new List<string> { inputText });
-            return response.FirstOrDefault();
+            Response<KeyPhraseCollection> results = await textAnalyticsClient.ExtractKeyPhrasesAsync(document, "en");
+
+            return results.Value;
         }
 
         /// <summary>
-        /// Gets sentiment score for the passed input text.
+        /// Gets sentiment score for the passed document.
         /// </summary>
-        /// <param name="inputText">Text to analyze</param>
-        public async Task<double?> GetSentimentAsync(string inputText)
-        {
-            var response = await GetSentimentBatchAsync(new List<string> { inputText });
-            return response.FirstOrDefault();
-        }
-
-        private async Task<IList<IList<string>>> GetKeyPhrasesBatchAsync(IList<string> batchInputText)
+        /// <param name="document">Text to analyze</param>
+        public async Task<double?> GetSentimentAsync(string document)
```

Comment:
What is the information you want to get from the document?
Currently you are returning the confidence score on the document having a neutral sentiment, and looking a the code, I don't think that is what you need.

Reply:
Yep, should not returning neutral sentiment of the confidence score. We should get the confidence score here.

Reply:
I want to make sure that the Object `ConfidenceScores` contains the information you actually need for this sample, and I don't see them being used in the code.
Just FYI, between versions 2 and 3 (current), Text Analytics changed the concept of the sentiment score. In v3, the `DocumentSentiment.Sentiment` contains the sentiment of the document, and the `DocumentSentiment.ConfidenceScores` contains the softmax value for each possible `Sentiment`.
Because of this relationship, the Confidence's scores are better use when next to the `Sentiment` label than by themselves.

More information => https://docs.microsoft.com/en-us/azure/cognitive-services/Text-Analytics/how-tos/text-analytics-how-to-sentiment-analysis?tabs=version-3-1#sentiment-analysis